RADIUS support (Closes #409)

pputerla opened the pull request:
<!-- Thanks for issuing a Pull Request (PR)! -->

### Description of PR
<!-- Describe your changes in detail -->

What component(s) does this PR affect?

- [ ] Back-End (Database, Detection/Configuration/etc. Microservices)
- [x] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs (incl. wiki)
- [ ] Build System

<!-- [Optional] Please elaborate on the affected component(s) here: -->

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Detection/Configuration/etc. Microservices)
- [ ] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs (incl. wiki)
- [ ] Build System

<!-- [Optional] Please elaborate on the component(s) requiring changes here: -->

### Related Issue
<!-- Please make sure you have an issue associated with this Pull Request -->
<!-- If you are suggesting a new feature or change, please discuss it in an issue first -->
<!-- If you are fixing a bug, there should be an issue describing it with steps to reproduce -->
<!-- Please don't forget to add `(close/fix #<issue-no>)` to the pull request title -->

<!-- Please link to the issue here: -->
Resolves #409 

### Solution
<!-- How is this issue solved/fixed? What is the main design/logic? -->
integration is inspired by LDAP integration (#108 ) 
radius connection is done via pyrad library (https://github.com/pyradius/pyrad)
conspect:
 * if enabled (`AUTH_METHOD = "radius"`): config validation is performed during bootstraping of the frontend
 * radius servers are tried out in the order provided in the config; last successful server is used for next attempt (caveat: for the sake of simplicity it is per gunicorn process)
 * for AccessAccept responses optional attribute inspection is done to support mapping to either "user" or "admin" role
 * user is not assigned password in the local db (i.e. RADIUS response is always required to log in)
 * superuser account is always available
### Type
<!--- What types of changes does your code introduce? -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

<!-- [Optional] If none of the above applies, please elaborate here -->

### Checklist:
<!-- Go over all the following points, and mark what applies. -->
- [x] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation.
- [x] I have updated the documentation accordingly.
_added self-documented entries in `webapp.cfg`_

pputerla commented on frontend/webapp/utils/radius/forms.py:
```diff
@@ -0,0 +1,126 @@
+
+        return True
+
+    # because the user manipulation is broken i.e. it can lead to multiple entries in roles_users table (perhaps lack of
```
@slowr 
I have discovered a bug in user manipulation code (tested on master).
To reproduce:
* create a user with user role
* delete this user from the UI
* create the same user (same username) with . create_user() method
* assign this user admin role using .add_role_to_user() method

now check the database; it will contain 2 entries in roles_users table for that user.

I don't know where should it be fixed. The natural place would be to add primary key constraint in roles_users table but I am not competenent enough to do that.
Other approach would be to remove user roles before removing him from the ui (actions_view.py)

Pls. can you comment on that?

slowr replied:
@pputerla I merged a fix for this. Can you verify that in your example there is no issue anymore?

pputerla commented:
Labeler is not working because the PR is from the forked repository

pputerla commented:
> LGTM 🎉
> 
> Thanks a lot for the PR and for adding support for RADIUS!
> 
> Please check if the fix that I introduced solves the `roles_users` problem and then we can merge this.

as a matter of fact it does solve the problem;
so now I have replaced sql with flask api calls: .remove_role_from_user() and .add_role_to_user() 

it's worth considering also adding primary key to the roles_users in models.py so new dbs will be protected against double entries in that table

slowr commented:
Thanks for the suggestion, I will add the key in a future release!
